src/java/src/main/java/ai/onnxruntime/genai/Tokenizer.java

```diff
@@ -79,6 +79,109 @@ public String[] decodeBatch(Sequences sequences) throws GenAIException {
     return result;
   }
 
+  /**
+   * Gets the beginning of sentence token ID.
+   *
+   * @return The BOS token ID.
+   * @throws GenAIException If the call to the GenAI native API fails.
+   */
+  public int getBosTokenId() throws GenAIException {
+    if (nativeHandle == 0) {
+      throw new IllegalStateException("Instance has been freed and is invalid");
+    }
+
+    return tokenizerGetBosTokenId(nativeHandle);
+  }
+
+  /**
+   * Gets the padding token ID.
+   *
+   * @return The padding token ID.
+   * @throws GenAIException If the call to the GenAI native API fails.
+   */
+  public int getPadTokenId() throws GenAIException {
+    if (nativeHandle == 0) {
+      throw new IllegalStateException("Instance has been freed and is invalid");
+    }
+
+    return tokenizerGetPadTokenId(nativeHandle);
+  }
+
+  /**
+   * Gets the end of sentence token IDs.
+   *
+   * @return An array of EOS token IDs.
+   * @throws GenAIException If the call to the GenAI native API fails.
+   */
+  public int[] getEosTokenIds() throws GenAIException {
+    if (nativeHandle == 0) {
+      throw new IllegalStateException("Instance has been freed and is invalid");
+    }
+
+    return tokenizerGetEosTokenIds(nativeHandle);
+  }
+
+  /**
+   * Converts a string to a token ID.
+   *
+   * @param str The string to convert to a token ID.
+   * @return The token ID for the given string.
+   * @throws GenAIException If the call to the GenAI native API fails.
+   */
+  public int toTokenId(String str) throws GenAIException {
+    if (nativeHandle == 0) {
+      throw new IllegalStateException("Instance has been freed and is invalid");
+    }
+
+    return tokenizerToTokenId(nativeHandle, str);
+  }
+
+  /**
+   * Applies a chat template to format messages.
+   *
+   * @param templateStr The template string to use.
+   * @param messages The messages in JSON format.
+   * @param tools The tools in JSON format (can be null).
+   * @param addGenerationPrompt Whether to add generation prompt.
+   * @return The formatted chat string.
+   * @throws GenAIException If the call to the GenAI native API fails.
+   */
+  public String applyChatTemplate(
+      String templateStr, String messages, String tools, boolean addGenerationPrompt)
+      throws GenAIException {
+    if (nativeHandle == 0) {
+      throw new IllegalStateException("Instance has been freed and is invalid");
+    }
+
+    return tokenizerApplyChatTemplate(
+        nativeHandle, templateStr, messages, tools, addGenerationPrompt);
+  }
+
+  /**
+   * Updates tokenizer options.
+   *
+   * @param options Map of option keys to values.
+   * @throws GenAIException If the call to the GenAI native API fails.
+   */
```

Comment:
should we document the currently supported options and their values here as well?

Reply:
No, I think just the C API is fine. Or else it'll be too much to maintain 😮‍💨